### DISCUSSION TITLE
MQTT not working with error message

### DISCUSSION THREAD

**Discussion by xvisor16** [zwave-js/zwave-js-ui#3208] [Q&A] [answered]
Hi Ya'll,

I want to migrate from openzwave to zwave-js-ui and got zwave-js-ui up and running on my Raspberry Pi 4B. Devices work but I can configure MQTT with any setting and it says it's oke but the log says the following:

2023-08-02 20:16:16.514 WARN Z-WAVE: Zwavejs driver is not ready yet, statistics will be enabled on driver initialization
2023-08-02 20:16:30.083 ERROR MQTT: Error while connecting MQTT Cannot read properties of undefined (reading 'connect')

All other things running on my Mosquitto broker are running fine. Mosquitto runs on the same machine but it doesn't matter what i put in localhost or the ip with or without mqtt:// the error is always the same.

I've tried googling the error to no avail and also can't find anything in the docs or github related to this error.

My broker is simple configured with and uses a username and password:
listener 1883
allow_anonymous false
password_file /etc/mosquitto/mospass

The mosquitto logs show all other mqtt devices no problem there:
1690999804: mosquitto version 2.0.12 running
1690999805: New connection from 192.168.68.52:40756 on port 1883.
1690999805: New client connected from 192.168.68.52:40756 as mqttjs_f6730c2a (p5, c1, k60, u'mqtt_user').
1690999805: New connection from 192.168.68.99:50590 on port 1883.
1690999805: New client connected from 192.168.68.99:50590 as mqtt-explorer-c670de45 (p2, c1, k60, u'mqtt_user').
1690999814: New connection from 192.168.68.97:33302 on port 1883.
1690999814: New client connected from 192.168.68.97:33302 as Domoticz-MQTT-AutoDiscover2cfb00c3-5d7b-42b9-a1b4-188c2814ab2522 (p2, c1, k120, u'mqtt_user').
1690999814: New connection from 192.168.68.97:33310 on port 1883.
1690999814: New client connected from 192.168.68.97:33310 as Domoticz-MQTT-AutoDiscover14dccf69-6f6f-4dfd-9626-366aeb232c5021 (p2, c1, k120, u'mqtt_user').

Zigbee2MQTT is running fine for instance.

Hope anyone has a suggestion to fix this.

Regards Peter

**Comment by AlCalzone** [comment]
MQTT is handled by zwave-js-ui. Transferring the discussion.
@robertsLando

**Reply by robertsLando** [reply, marked as answer]
Pull from master, I have updated mqtt library but I haven't done a release yet so I think you are using master tag in docker or you are manually building using git/node so in both cases just pull from master

**Reply by xvisor16** [reply]
Ohw Man, that just cost me a day 🤣
It's working now 🤭

**Reply by robertsLando** [reply]
@xvisor16 I'm sorry for that, anyway I suggest you to only use stable releases, master can have problems :)

**Reply by xvisor16** [reply]
@robertsLando, No problem and I know, but I do like to work with git for software without installer/updater. So when I saw the 8.22.0 just released on monday and the fact I always use a dev branch for my own active work. I made the mistake to take master for a working version without checking for source changes. But all is well now and working and no longer using openzwave which is a big plus since zwave-js is sooo much faster. Thanks!
